packages/oasis-actions/src/protocols/aave/getAaveProtocolData.ts

```diff
@@ -80,35 +85,52 @@ export const getAaveProtocolData = async ({
     aaveProtocolDataProvider.getReserveConfigurationData(collateralTokenAddress),
   ]
 
-  if (isV3) {
-    promises.push(aaveProtocolDataProvider.getReserveEModeCategory(collateralTokenAddress))
-  }
-
   if (hasProxy) {
     promises.push(aaveProtocolDataProvider.getUserReserveData(debtTokenAddress, proxy))
     promises.push(aaveProtocolDataProvider.getUserReserveData(collateralTokenAddress, proxy))
   }
 
+  if (isV3) {
+    // TODO: Get reserveEModeCategory for debt token and compare
+    promises.push(aaveProtocolDataProvider.getReserveEModeCategory(collateralTokenAddress))
+    promises.push(aaveProtocolDataProvider.getReserveEModeCategory(debtTokenAddress))
+  }
+
   const results = await Promise.all(promises)
 
-  const reserveEModeCategory = isV3 && results[5] ? Number(results[5].toString()) : undefined
+  let reserveEModeCategory
+  const doesNotHaveProxy = !hasProxy
+  if (hasProxy && results[7] && results[8]) {
+    const collateralEModeCategory = Number(results[7].toString())
+    const debtEModeCategory = Number(results[8].toString())
+    reserveEModeCategory =
+      collateralEModeCategory === debtEModeCategory ? collateralEModeCategory : 0
+  }
+  if (doesNotHaveProxy && results[5] && results[6]) {
+    const collateralEModeCategory = Number(results[5].toString())
+    const debtEModeCategory = Number(results[6].toString())
+    reserveEModeCategory =
+      collateralEModeCategory === debtEModeCategory ? collateralEModeCategory : 0
+  }
 
   let eModeCategoryData
   if (isV3 && aavePool && reserveEModeCategory !== 0) {
     eModeCategoryData = await aavePool.getEModeCategoryData(reserveEModeCategory)
   }
 
-  return {
+  const protocolData = {
```

Comment:
Unnecessary variable.
Please add types here. It's `any` everywhere. At least add type for returned value.

Reply:
You've sorted this out in your PR